Scripts/RemediationScripts/Remediate-EnableMicrosoftDefenders.ps1

```diff
@@ -270,7 +451,7 @@ function Enable-MicrosoftDefender
     $currentLoginRoleAssignments = Get-AzRoleAssignment -SignInName $context.Account.Id -Scope "/subscriptions/$($SubscriptionId)";
     $roles = $currentLoginRoleAssignments | Where { ($_.RoleDefinitionName -eq "Owner" -or $_.RoleDefinitionName -eq "Security Admin" ) -and !($_.Scope -like "/subscriptions/$($SubscriptionId)/resourceGroups")}
 
-    if(($currentLoginRoleAssignments | Where { ($_.RoleDefinitionName -eq "Owner" -or $_.RoleDefinitionName -eq "Security Admin" ) -and !($_.Scope -like "/subscriptions/$($SubscriptionId)/resourceGroups")} | Measure-Object).Count -le 0)
+    if(($roles | Where { ($_.RoleDefinitionName -eq "Owner" -or $_.RoleDefinitionName -eq "Security Admin" ) -and !($_.Scope -like "/subscriptions/$($SubscriptionId)/resourceGroups")} | Measure-Object).Count -le 0)
```

Comment:
why are we checking same condition again?

Reply:
Updated if statement to avoid checking same condition